sdk/core/Azure.Core.Experimental/src/Primitives/BinaryData.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Azure.Core.Experimental.Primitives
+{
+    /// <summary>
+    /// Abstraction for a payload of binary data.
+    /// </summary>
+    public class BinaryData
+    {
+        /// <summary>
+        /// The backing store for the <see cref="BinaryData"/> instance.
+        /// </summary>
+        internal ReadOnlyMemory<byte> Memory { get; }
```

Comment:
~~This is so much of a nit that I don't actually want you to change anything, but it somehow bothers me the tiniest bit that we're not calling it `Data` since it's the `BinaryData` class.~~

Reply:
Updated!